benchmarks/hash_table/static_multimap/insert_bench.cu

```diff
@@ -14,44 +14,50 @@
  * limitations under the License.
  */
 
+#include <defaults.hpp>
 #include <key_generator.hpp>
 
 #include <cuco/static_multimap.cuh>
 
 #include <nvbench/nvbench.cuh>
 
 #include <thrust/device_vector.h>
+#include <thrust/execution_policy.h>
+#include <thrust/transform.h>
+
+namespace cuco {
+namespace benchmark {
 
 /**
- * @brief A benchmark evaluating multi-value `insert` performance:
- * - Total number of insertions: 100'000'000
- * - CG size: 8
+ * @brief A benchmark evaluating multi-value `insert` performance
  */
-template <typename Key, typename Value, dist_type Dist, nvbench::int32_t Multiplicity>
-std::enable_if_t<(sizeof(Key) == sizeof(Value)), void> nvbench_static_multimap_insert(
-  nvbench::state& state,
-  nvbench::type_list<Key, Value, nvbench::enum_type<Dist>, nvbench::enum_type<Multiplicity>>)
+template <typename Key, typename Value, typename Dist>
+std::enable_if_t<(sizeof(Key) == sizeof(Value)), void> static_multimap_insert(
+  nvbench::state& state, nvbench::type_list<Key, Value, Dist>)
 {
+  using pair_type = cuco::pair_type<Key, Value>;
+
   auto const num_keys  = state.get_int64("NumInputs");
   auto const occupancy = state.get_float64("Occupancy");
 
   std::size_t const size = num_keys / occupancy;
 
-  std::vector<Key> h_keys(num_keys);
-  std::vector<cuco::pair_type<Key, Value>> h_pairs(num_keys);
-
-  generate_keys<Dist, Multiplicity, Key>(h_keys.begin(), h_keys.end());
+  thrust::device_vector<Key> keys(num_keys);
 
-  for (auto i = 0; i < num_keys; ++i) {
-    Key key           = h_keys[i];
-    Value val         = h_keys[i];
-    h_pairs[i].first  = key;
-    h_pairs[i].second = val;
-  }
+  key_generator gen;
+  gen.generate<Dist>(state, thrust::device, keys.begin(), keys.end());
 
-  thrust::device_vector<cuco::pair_type<Key, Value>> d_pairs(h_pairs);
+  thrust::device_vector<pair_type> pairs(num_keys);
+  thrust::transform(thrust::device,
+                    keys.begin(),
+                    keys.end(),
+                    pairs.begin(),
+                    [] __host__ __device__(Key const& key) {
+                      return thrust::raw_reference_cast(pair_type(key, 42));
+                    });
```

Comment:
```suggestion
  auto pairs_begin = thrust::make_transform_iterator(
      keys.begin(),
      [] __device__(auto i) { return pair_type(i, i); });
```
IIRC, `transform` takes a long time to compile and transform iterator can avoid materializing the data.

Reply:
Turns out there's on average <1% diff in timings but compile times are only 3% faster with `transform` compared to using `transform_iterator`